pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -110,6 +109,35 @@ func (r *AtlasDataFederationReconciler) Reconcile(contextInt context.Context, re
 	}
 	ctx.Client = atlasClient
 
+	owner, err := customresource.IsOwner(dataFederation, r.ObjectDeletionProtection, customresource.IsResourceManagedByOperator, managedByAtlas(contextInt, atlasClient, project.ID(), log))
+	if err != nil {
+		result = workflow.Terminate(workflow.Internal, fmt.Sprintf("unable to resolve ownership for deletion protection: %s", err))
+		ctx.SetConditionFromResult(status.DataFederationReadyType, result)
+		log.Error(result.GetMessage())
+
+		return result.ReconcileResult(), nil
+	}
+
+	if !owner {
+		result = workflow.Terminate(
+			workflow.AtlasDeletionProtection,
+			"unable to reconcile DataFederation: it already exists in Atlas, it was not previously managed by the operator, and the deletion protection is enabled.",
+		)
+		ctx.SetConditionFromResult(status.DataFederationReadyType, result)
+		log.Error(result.GetMessage())
+
+		return result.ReconcileResult(), nil
+	}
+
+	err = customresource.ApplyLastConfigApplied(contextInt, project, r.Client)
+	if err != nil {
+		result = workflow.Terminate(workflow.Internal, err.Error())
+		ctx.SetConditionFromResult(status.DataFederationReadyType, result)
+		log.Error(result.GetMessage())
+
+		return result.ReconcileResult(), nil
+	}
```

Comment:
Maybe it is a bit early for this piece here, I mean it should probably be set once nothing else is expected to mutate the k8s object, and I see a lot of logic below that could potentially be making changes.

Should we move this below, maybe after deletion?